.github/workflows/dep-update-review.yml

```diff
@@ -0,0 +1,63 @@
+name: Dependency Update Reviewer
+
+on:
+  pull_request:
+    types:
+      - opened
+      - synchronize
+      - reopened
+    paths:
+      - "**Cargo.lock"
+      - "**Cargo.toml"
+
+jobs:
+  generate-report:
+    runs-on: ubuntu-latest-xl
+    name: Comment update review report on PR
+    env:
+      MAX_ALLOWED_CHARS: 65536
+    steps:
+      - name: checkout
+        uses: actions/checkout@v2.3.4
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+      - name: checkout base ref
+        uses: actions/checkout@v2.3.4
+        with:
+          ref: ${{ github.event.pull_request.base.sha }}
+          path: diem-base
+      - name: install depdive
+        run: cargo install --git https://github.com/diem/whackadep --rev 9dbcadc37a6367e59dc5729224cb2a8f46907360 depdive
+      - name: Get depdive update report
+        id: set-update-report
+        run: |
+          output="$(depdive update-review paths ./diem-base/ ./)"
+          echo "${output}"
+          output="${output//'%'/'%25'}"
+          output="${output//$'\n'/'%0A'}"
+          output="${output//$'\r'/'%0D'}"
+          echo "::set-output name=report::${output}"
+          length=$(${output} | wc -c)
+          echo "${length}"
+          flag=$(( ${length} < ${MAX_ALLOWED_CHARS} ))
+          echo "::set-output name=flag::${flag}"
+          echo "${flag}"
+      - name: make valid comment
+        if: ${{ steps.set-update-report.outputs.report && steps.set-update-report.outputs.flag == 1 }}
+        uses: diem/actions/comment@a5fa31ff2b54a544cd88ac912cda469742592e6c
+        with:
+          comment: |
+            ${{ steps.set-update-report.outputs.report }}
+          tag: update-review-report
+          delete-older: true
+      - name: make comment when output too big
+        if: ${{ steps.set-update-report.outputs.flag == 0 }}
+        uses: diem/actions/comment@a5fa31ff2b54a544cd88ac912cda469742592e6c
+        with:
+          # todo if comment too big, create a gist and post the gist link
+          comment: |
+            `This PR contains dependency updates.
+            However, Depdive update review report exceeds max. characters allowed in a GitHub comment.
+            Please, run depdive locally or check actions.`
+          tag: update-review-report
+          delete-older: true
```

Comment:
Instead of having two different steps for if the output is too big or not I would probably set the comment accordingly so that you only need to have a single comment step.

Reply:
@nasifimtiazohi thoughts on this?

Reply:
Sorry I somehow missed this comment. We can do an `if-else` condition to set output in this way. Originally my rationale was to revise this step to implement creating a gist and posting the link of the gist to the comment using this [action](https://github.com/marketplace/actions/deploy-to-gist). However, we'd need an appropriate token for that which is why I kept it to do.
My rationale when revising, this way, we'd only need to revise this specific step and won't have to bother with the rest.